### ISSUE TITLE
Composer installer

### ISSUE THREAD

**Issue by prhost** [prhost/new-opencart#10]
Adoro o composer, acho que foi uma das coisas mais perfeita ja criada no mundo PHP. Uso ele para gerenciar e atualizar todo o projeto em qualquer ambiente, dev, test e production.
    Existe um projeto do composer chamado Installers https://github.com/composer/installers, nele a galera adiciona "type" que servem para indicar o que deve ser feito com o pacote instalado que tiver especifico o mesmo "type". Deem uma olhada no repositório deles se tiverem dúvidas. 
    Atualmente criei um installers para gerenciar "core", "extensions" e "themes", assim cada um deles fica separado em repositórios, se eu quiser usar um determinado tema, eu simplesmente do um `composer require nome-do-tema` e ele automaticamente adiciona o tema na pasta `themes/`. O mesmo acontece com o core que é adicionado na pasta `app/` e as extensões na pasta `extensions/`.

![oc_-_composer_installer_1](https://cloud.githubusercontent.com/assets/3984760/12620703/791516b4-c504-11e5-9f6e-ef276c641f95.jpg)

![oc - composer installer 2](https://cloud.githubusercontent.com/assets/3984760/12620706/7bc1f08a-c504-11e5-975a-b9a898f6ee36.jpg)

Desculpe ocultar algumas coisas, são dados da empresa onde trabalho.

**Comment by stefanhuber** [comment]
Unfortunately, this is only Portuguese. 

I have done two projects in this direction:
- Composer installer for extensions: https://github.com/beyondit/opencart-extension-installer
- Composer project template: https://github.com/beyondit/opencart-project-template

Maybe there is some interest in collaboration.

**Comment by prhost** [comment]
@stefanhuber like many of his projects. I began to develop much of this project but is in beta, not yet divulged. I liked your project related to the composer, my was to use the composer to manage themes and extensions as external packages, easy to add and remove. If you are interested we can do something together.

**Comment by stefanhuber** [comment]
@prhost Yes I think managing all kind of extensions with composer is the way to go. Many things are possible in this regard. The extension installer i created is working basically. Maybe we can share some thoughts via email, how we could do something. Please send me a mail, because I cannot find your address anywhere...

**Comment by prhost** [comment]
@stefanhuber i enabled in my profile, see there
